openedx/core/djangoapps/credit/tasks.py

```diff
@@ -33,23 +33,94 @@ def update_course_requirements(course_id):
         is_credit_course = CreditCourse.is_credit_course(course_key)
         if is_credit_course:
             course = modulestore().get_course(course_key)
-            requirements = [
-                {
-                    "namespace": "grade",
-                    "name": "grade",
-                    "criteria": {
-                        "min_grade": get_min_grade_for_credit(course)
-                    }
-                }
-            ]
+            requirements = _get_course_credit_requirements(course)
             set_credit_requirements(course_key, requirements)
-    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements) as exc:
+    except (InvalidKeyError, ItemNotFoundError, InvalidCreditRequirements, AttributeError) as exc:
```

Comment:
I'd prefer to handle this closer to where the `AttributeError` might be raised so we can provide a more detailed error message.